apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -509,15 +509,14 @@ impl FetchDependencyGraphNodePath {
         type_conditioned_fetching_enabled: bool,
         root_type: CompositeTypeDefinitionPosition,
     ) -> Result<Self, FederationError> {
-        let root_possible_types = if type_conditioned_fetching_enabled {
+        let root_possible_types: IndexSet<Name> = if type_conditioned_fetching_enabled {
             schema.possible_runtime_types(root_type)?
         } else {
             Default::default()
         }
         .into_iter()
-        .map(|pos| Ok(pos.get(schema.schema())?.name.clone()))
```

Comment:
Bugs:
- This should be a sorted list, similar to the JS code. (I've also updated the code to use `process_results()`.)

Reply:
That's great. It will keep semantic diff simple.